bip-xxxx.mediawiki

```diff
@@ -0,0 +1,303 @@
+
+===Preventing mempool replacement===
+A safe way to implement payjoin, is for both the sender and receiver to try broadcasting the original transaction at some fixed interval period regardless of the state of the payjoin.
+
+If the receiver was not properly adding fees to the payjoin transaction, the original transaction would end up replacing the payjoin transaction in the mempool.
+
+===Defeating heuristics based on the fee calculation===
+Most wallets are creating a round fee rate (like 2 sat/b).
+If the payjoin transaction's fee was not increased by the added size, then those payjoin transaction could easily be identifiable on the blockchain.
+
+Not only would those transactions would stand out by not having a round fee (like 1.87 sat/b), but any suspicion of payjoin could be confirmed by checking if removing one input would create a round fee rate.
+
+===Receiver does not need to be a full node===
+Because the receiver needs to bump the fee to keep the same fee rate as the original PSBT, it needs the input's UTXO information to know what is the original fee rate. Without PSBT, light wallets like Wasabi Wallet would not be able to receive a payjoin transaction.
+
+The validation (policy and consensus) of the original transaction is optional: a receiver without a full node can decide to create the payjoin transaction and automatically broadcast the original transaction after a timeout of 1 minute, and only verifying that it has been propagated in the network.
+
+However, automated systems (like BTCPay Server) need to verify the transaction to prevent UTXO probing attacks. 
+
+This is not a concern for receivers using end-user wallets such as Wasabi Wallet, such receivers can just limit the number of original PSBT proposals of a specific address to one. With such wallets, the attacker has no way to generate new deposit addresses to probe the UTXOs.
+
+===Spare change donation===
+Small change inside wallets are detrimental to privacy. Mixers like Wasabi wallet, because of its protocol, eventually generate such [[https://docs.wasabiwallet.io/using-wasabi/ChangeCoins.html#spend-the-change-with-another-entity-where-you-don-t-mind-if-each-of-the-two-knows-that-you-transact-with-the-other-entity|small change]].
+
+A common way to protect your privacy is to donate those spare changes, to deposit them in an exchange or on your favorite merchant's store account. Those kind of transactions can easily be spotted on the blockchain: There is only one output.
+
+However, if you donate via payjoin, it will look like a normal transaction.
+
+On top of this the receiver can poison analysis by randomly faking a round amount of satoshi for the additional output.
+
+===Impacted heuristics===
+Our proposal of payjoin is breaking the following blockchain heuristics:
+
+* Common inputs heuristics.
+
+Because payjoin is mixing the inputs of the sender and receiver, this heuristic becomes unreliable.
```

Comment:
In case the receiver does not add any inputs then the sender could use the original transaction instead of the proposed by the receiver because is is hard to believe the sender can improve his privacy in this way and there is a risk of the receiver is sending to an exchange, betting site or something like that.

Reply:
Sender can increase privacy even if inputs are not added, for example, if the receiver switched his output from P2SH to P2WSH to match the type of inputs.